frontend/server/src/DAO/Identities.php

```diff
@@ -107,6 +107,81 @@ public static function findByUserId(int $userId): ?\OmegaUp\DAO\VO\Identities {
         return new \OmegaUp\DAO\VO\Identities($rs);
     }
 
+    /**
+     * @return array{identity: \OmegaUp\DAO\VO\Identities, isMainIdentity: bool, loggedAsMainIdentity: bool}|null
+     */
+    public static function resolveAssociatedIdentity(
+        string $usernameOrEmail,
+        \OmegaUp\DAO\VO\Identities $currentIdentity
+    ): ?array {
+        $sql = 'SELECT
+                    i.*,
+                    aut.acting_identity_id,
+                    aut.user_id AS acting_user_id,
+                    u.main_identity_id,
+                    e.email
+                FROM
+                    Identities i
+                INNER JOIN
+                    Users u
+                ON
+                    i.user_id = u.user_id
+                INNER JOIN
+                    Emails e
+                ON
+                    e.email_id = u.main_email_id
+                LEFT JOIN
+                    Auth_Tokens aut
+                ON
+                    aut.user_id = i.user_id
+                WHERE
+                    i.user_id = (
```

Comment:
```suggestion
                    i.user_id IN (
```

como una subconsulta puede tener una cantidad arbitraria de registros, se debería usar un `IN` en vez de `=`.

habiendo dicho eso, si ya se está pasando la identidad actual, esta subconsulta es necesaria? ya sabemos que `i.user_id` _debería_ ser igual a `$currentIdentity->user_id`. lo único que deberíamos checar entonces es que el `i.username = ${usernameOrEmail}` (las identidades asociadas no tienen correos, entonces podemos quitarle el "or email" a la variable) y que `i.user_id = ${currentIdentity->user_id}`.

Reply:
Cierto, ya no se requiere la subconsulta. No sé porqué estaba pensando que las identidades asociadas no tenían el `user_id`, pero justamente ese es el campo que las asocia.